components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/OAuthUtil.java

```diff
@@ -693,36 +704,98 @@ public static boolean revokeAuthzCodes(String username, UserStoreManager userSto
+                String primaryOrganizationId = OAuthComponentServiceHolder.getInstance().getOrganizationManager()
+                        .getPrimaryOrganizationId(accessingOrg);
+                tenantDomain = OAuthComponentServiceHolder.getInstance().getOrganizationManager()
+                        .resolveTenantDomain(primaryOrganizationId);
+                IdentityProvider orgSsoIdp = OAuthComponentServiceHolder.getInstance().getIdpManager()
+                        .getIdPByRealmId(ORGANIZATION_LOGIN_HOME_REALM_IDENTIFIER, tenantDomain);
+                if (orgSsoIdp != null) {
+                    authenticatedUser.setFederatedIdPName(orgSsoIdp.getIdentityProviderName());
+                }
+                authenticatedUser.setUserResidentOrganization(accessingOrg);
+            }
+            authenticatedUser.setAccessingOrganization(accessingOrg);
+            authenticatedUser.setTenantDomain(tenantDomain);
+            return authenticatedUser;
+        } catch (OrganizationManagementException e) {
+            LOG.error("Error occurred while resolving organization information for the tenant : " + tenantDomain);
+            throw new UserStoreException(e);
+        } catch (IdentityProviderManagementException e) {
+            LOG.error("Error occurred while resolving IDP name of the organization login IDP in : " + tenantDomain);
+            throw new UserStoreException(e);
+        }
     }
 
     /**
      * Get clientIds of associated application of an application role.
      * @param role          Role object.
-     * @param tenantDomain  Tenant domain.
+     * @param authenticatedUser  Authenticated user.
      * @return Set of clientIds of associated applications.
      */
-    private static Set<String> getClientIdsOfAssociatedApplications(Role role, String tenantDomain) {
+    private static Set<String> getClientIdsOfAssociatedApplications(Role role, AuthenticatedUser authenticatedUser) {
 
         ApplicationManagementService applicationManagementService =
                 OAuthComponentServiceHolder.getInstance().getApplicationManagementService();
-        List<AssociatedApplication> associatedApplications = role.getAssociatedApplications();
+        List<String> associatedApplications = role.getAssociatedApplications().stream()
+                .map(AssociatedApplication::getId).collect(Collectors.toList());
+        try {
+            if (authenticatedUser.getUserResidentOrganization() != null) {
+                List<String> newAssociatedApplications = new ArrayList<>();
+                for (String app : associatedApplications) {
+                    newAssociatedApplications.add(
+                            SharedAppResolveDAO.getMainApplication(app, authenticatedUser.getAccessingOrganization()));
+                }
+                associatedApplications = newAssociatedApplications;
+            }
+        } catch (IdentityOAuth2Exception e) {
+            throw new RuntimeException(e);
```

Comment:
Fix this

Reply:
Fixed by - https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/2382/commits/401d9c67e6831c1ae56ccafb4d004920242e713c